packages/py-moose-lib/moose_lib/utilities/structured_logging.py

```
@@ -0,0 +1,97 @@
+# Context variable to store the current logging context
+_logging_context: ContextVar[Optional[Dict[str, Any]]] = ContextVar(
+    "_logging_context", default=None
+)
+
+
+class StructuredLogContext:
+    """
+    Context manager for structured logging with propagated context.
+    """
+
+    def __init__(
```

Ruff (0.14.14)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)